magda-storage-api/src/MagdaMinioClient.ts

```diff
@@ -0,0 +1,122 @@
+import ObjectFromStore from "./ObjectFromStore";
+import ObjectStoreClient from "./ObjectStoreClient";
+import { Stream, Readable } from "stream";
+
+import * as Minio from "minio";
+
+export default class MagdaMinioClient implements ObjectStoreClient {
+    private readonly client: any;
+    private readonly bucketName: string = "magda-bucket";
+
+    constructor({ endPoint, port, useSSL, accessKey, secretKey }: any) {
+        this.client = new Minio.Client({
+            endPoint,
+            port,
+            useSSL,
+            accessKey,
+            secretKey
+        });
+        this.client.makeBucket(this.bucketName, "us-east-1", (err: Error) => {
```

Comment:
- What's the point of making a bucket on startup now that the caller of the API specified a bucket?
- Given that the caller of the API specifies the bucket, how do they make sure it's created before they put something in it?
- Should `"us-east-1"` be configurable?

Reply:
> What's the point of making a bucket on startup now that the caller of the API specified a bucket?

What we discussed was that as a proof of concept, we will start with a generic bucket where everything is stored, and then move on to a bucket for each service. This is the generic bucket in question. The caller of the API needs to use the generic bucket's name for the moment in their request until the storage service starts keeping track of who owns which bucket.

> Given that the caller of the API specifies the bucket, how do they make sure it's created before they put something in it?

As things are, they would get a relevant error.

> Should "us-east-1" be configurable?
Yes. I just made it so :+1:

Reply:
OK 👍